File: OLSource1/v1/VS_officedev/codesnippet/CSharp/walkthrough--updating-the-controls-on-a-ribbon-at-run-time_3.cs
        private RibbonDropDownItem CreateRibbonDropDownItem()
        {
            return this.Factory.CreateRibbonDropDownItem();
        }
        private RibbonMenu CreateRibbonMenu()
        {
            return this.Factory.CreateRibbonMenu();
        }
        private RibbonButton CreateRibbonButton()
        {
            RibbonButton button = this.Factory.CreateRibbonButton();
            button.Click += new RibbonControlEventHandler(button_Click);
            return button;
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/walkthrough--updating-the-controls-on-a-ribbon-at-run-time_3.vb

    Private Function CreateRibbonDropDownItem() As RibbonDropDownItem
        Return Me.Factory.CreateRibbonDropDownItem()
    End Function

    Private Function CreateRibbonMenu() As RibbonMenu
        Return Me.Factory.CreateRibbonMenu()
    End Function

    Private Function CreateRibbonButton() As RibbonButton
        Dim button As RibbonButton = Me.Factory.CreateRibbonButton()
        AddHandler (button.Click), AddressOf Button_Click
        Return button
    End Function